src/util/system.h

```diff
@@ -263,6 +262,27 @@ class ArgsManager
      */
     std::optional<const Command> GetCommand() const;
 
+    /**
+     * Get blocks directory path
+     *
+     * @return Blocks path which is network specific
+     */
+    const fs::path& GetBlocksDirPath();
+
+    /**
+     * Get data directory path
+     *
+     * @param fNetSpecific Append network identifier to the returned path
+     * @return Absolute path on success, otherwise an empty path when a non-directory path would be returned
+     * @post Returned directory path is created unless it is empty
+     */
+    const fs::path &GetDataDirPath(bool fNetSpecific = true);
+
+    /**
+     * For testing
+     */
+    void ClearPathCache();
```

Comment:
The description of this function is not true: https://github.com/bitcoin/bitcoin/blob/834b971bff371951e3e63617d0a3fb9b3ad79c62/src/util/system.cpp#L974

Reply:
Do you have a suggestion for the description?

I think the comment is mostly unhelpful as it is too generic anyway. I would write something like that it's useful to propagate changes in args. I'm not sure about proper English wording though.

Reply:
Obviously, I'm not an [Official Namer Of Stuff](https://github.com/bitcoin/bitcoin/pull/20861#discussion_r596389178) :)

Maybe "Clean cached directory paths"?

Reply:
Will be addressed in the next push.